### PR TITLE
CI Workflow Dependency Upgrades + Optimization 👷

### PR DESCRIPTION
**Pull request by ndegwamartin**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes [link to issue]

**Engineer Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [x] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Review by pld — DISMISSED** [review]

**Comment by pld** [comment]
this is the main error I see in the quest logs

```
2024-07-10T18:55:27.7610798Z Error: Failed to generate Emma/JaCoCo coverage. Is Emma/JaCoCo jar on classpath?
```

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,7 +18,7 @@ env:
 
 jobs:
   engine-tests:
-    runs-on: macos-13
+    runs-on: ubuntu-latest
```
Comment (ellykits) [inline .github/workflows/ci.yml:21]: Does the Ubuntu image include an emulator? The macOS image was used to allow us to run instrumentation tests.

Comment (ellykits) [reply .github/workflows/ci.yml:21]: The performance tests require an emulator to run.

Comment (pld) [reply .github/workflows/ci.yml:21]: looks like yes, https://github.com/marketplace/actions/android-emulator-runner

Comment (ndegwamartin) [reply .github/workflows/ci.yml:21]: Yeah and its actually much faster. Preliminary tests show around 50% improvement for the jobs

**Comment by ndegwamartin** [comment]
> this is the main error I see in the quest logs
> 
> ```
> 2024-07-10T18:55:27.7610798Z Error: Failed to generate Emma/JaCoCo coverage. Is Emma/JaCoCo jar on classpath?
> ```
This is caused by disabling the instrumentation tests in the gradle configuration file. When disabled the reporting artifacts are not loaded. Note, we can only enable it after resolving  #3055

**Review by pld — APPROVED** [review]

**Pull request merged**